projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts

```diff
@@ -1,27 +1,45 @@
-import { Injectable } from '@angular/core';
+import {
+  Injectable,
+  Type,
+  Optional,
+  Inject,
+  forwardRef,
+  InjectionToken
+} from '@angular/core';
+import { LazyElementModuleRootOptions } from './lazy-elements.module';
 
 const LOG_PREFIX = '@angular-extensions/elements';
 
+export const LAZY_ELEMENT_ROOT_OPTIONS = new InjectionToken<
+  LazyElementModuleRootOptions
+>('LAZY_ELEMENT_ROOT_OPTIONS');
+
 export interface ElementConfig {
   tag: string;
   url: string;
   isModule?: boolean;
+  loadingComponent?: Type<any>;
+  errorComponent?: Type<any>;
 }
 
-@Injectable({
-  providedIn: 'root'
-})
+@Injectable()
 export class LazyElementsLoaderService {
   registry: Map<string, Promise<void>> = new Map<string, Promise<void>>();
   configs: ElementConfig[] = [];
 
-  constructor() {}
+  constructor(
+    @Optional()
+    @Inject(forwardRef(() => LAZY_ELEMENT_ROOT_OPTIONS))
```

Comment:
Why is here the `forwardRef` is it needed ? Because the toke in declared just above so it should be available also without forward ref ?